src/relay/backend/te_compiler_cache.cc

```diff
@@ -1088,6 +1088,30 @@ std::tuple<Array<te::Tensor>, Array<runtime::NDArray>, std::string> LowerTECompu
   return std::make_tuple(tensor_outs, constants, lower_te_compute.candidate_name_);
 }
 
+std::pair<Optional<tir::PrimFunc>, std::string> LowerToPrimFunc(const Function& relay_func,
```

Comment:
Should we add a check in here that relay_func has attr::kPrimitive to alert the user early if they are trying to lower a non-primitive function?

Reply:
Thanks, yes we should do that. Updated.